### PR TITLE
🐶📦 Pack the dogs: Add docker build for identical outputs

### PR DESCRIPTION
**Pull request by greg-nagy**
CleanShot 2022-04-28 at 18 45 15

# Why?
Firefox requires uploading source code if
minifier is used and to be able to compile
identical output to the uploaded package.
Our builds are environment dependent at the
moment because of the minification and source
map process. Long term solution will be to upgrade
our build process to be able to produce identical
file assets, but until then we use Docker.

# What, How, and Test

See README

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM node:16-alpine
+
+WORKDIR /tally-ho
+
+COPY . . 
+
+RUN mv .env.prod .env
+RUN apk add --no-cache python3 py3-pip git make bash && ln -sf python3 /usr/bin/python
+# sqlite compile throws an error during install, but it does not cause any problem so we are ignoring it
+RUN yarn install --frozen-lockfile || true
```
[the remark this answers is not in the document]
Comment (greg-nagy) [reply Dockerfile:10]: Yeah, I didn't like it either, but couldn't find a better way to do it. 

On the other hand, I think if something "important" is broken, the build step will fail, and that's good enough for us.

**Comment by Shadowfiend** [comment]
> Also - I wonder if the extension repo is the best place for this code?

If it's part of the build or release process it should probably stay in this repo.

**Comment by greg-nagy** [comment]
>> Also - I wonder if the extension repo is the best place for this code?

> If it's part of the build or release process it should probably stay in this repo.

So the thing is that I really don't like that we need this. But at the moment this is the best idea we have to solve the issue. 
Any insight is appreciated, and more than welcome. 

For reference here is a diff between our prod build from the gh release page and a local build from the same tag version of the code with the prod API keys supplied. It seems that we still have paths in the sourceURLs. 

Note: You can search for `diff /Users/greg/Downloads/` to see what files has changed.

**Review by 0xDaedalus — APPROVED** [review]
Lets 🚢

**Pull request merged**